torch/csrc/cudnn/BatchNorm.cpp

```diff
@@ -8,40 +8,37 @@ namespace torch { namespace cudnn {
 
 namespace {
 
-void setInputDescriptor(TensorDescriptor& desc, cudnnDataType_t dataType, THVoidTensor* tensor)
+void setInputDescriptor(TensorDescriptor& desc, cudnnDataType_t dataType, const at::Tensor& tensor)
 {
-  CHECK_ARG(tensor->nDimension >= 2 && tensor->nDimension <= 5);
+  CHECK_ARG(tensor.dim() >= 2 && tensor.dim() <= 5);
   int inputSize[5] = {0};
   int inputStride[5] = {0};
-  int nDimension = (tensor->nDimension <= 4) ? 4 : tensor->nDimension;
-  for (int i = 0; i < tensor->nDimension; ++i) {
-    inputSize[i] = (int) tensor->size[i];
-    inputStride[i] = (int) tensor->stride[i];
+  int nDimension = (tensor.dim() <= 4) ? 4 : tensor.dim();
+  for (int i = 0; i < tensor.dim(); ++i) {
+    inputSize[i] = (int) tensor.size(i);
+    inputStride[i] = (int) tensor.stride(i);
   }
-  for (int i = tensor->nDimension; i < nDimension; ++i) {
+  for (int i = tensor.dim(); i < nDimension; ++i) {
     inputSize[i] = 1;
     inputStride[i] = 1;
   }
   desc.set(dataType, nDimension, inputSize, inputStride);
 }
 
-void setScaleDescriptor(TensorDescriptor& desc, cudnnDataType_t dataType, THVoidTensor* tensor, int nDim)
+void setScaleDescriptor(TensorDescriptor& desc, cudnnDataType_t dataType, const at::Tensor& tensor, int nDim)
 {
-  CHECK_ARG(tensor->nDimension == 1);
-  CHECK_ARG(tensor->stride[0] == 1);  // scale must be contiguous
-  int size = (int) tensor->size[0];
-  int stride = (int) tensor->stride[0];
+  CHECK_ARG(tensor.dim() == 1);
+  CHECK_ARG(tensor.stride(0) == 1);  // scale must be contiguous
+  int size = (int) tensor.size(0);
+  int stride = (int) tensor.stride(0);
   int inputSize[5] = { 1, size, 1, 1, 1 };
   int inputStride[5] = { size * stride, stride, 1, 1, 1 };
   desc.set(dataType, (nDim <= 4) ? 4 : 5, inputSize, inputStride);
 }
 
-void* tensorPointer(cudnnDataType_t dataType, THVoidTensor* tensor)
+void* tensorPointer(cudnnDataType_t dataType, const at::Tensor& tensor)
 {
-  int elementSize = dataSize(dataType);
-  char* ptr = (char*) tensor->storage->data;
-  ptr += elementSize * tensor->storageOffset;
-  return ptr;
+  return tensor.data_ptr();
```

Comment:
Can't we just replace all calls with `.data_ptr()`?

Reply:
Yes.